hack/update_codegen.sh

```diff
@@ -15,7 +15,12 @@ if [ -d "$CODE_GEN_DIR" ]; then
     echo "Using cached code generator version: $VERSION"
 else
     git clone https://github.com/kubernetes/code-generator.git "${CODE_GEN_DIR}"
-    cd "${CODE_GEN_DIR}" && git reset --hard "${VERSION}" && go mod init && cd -
+    cd "${CODE_GEN_DIR}"
+    git reset --hard "${VERSION}"
+    if [[ ! -f go.mod ]]; then
+        go mod init
```

Comment:
I'm wondering why we need this at all, given that `go.mod` is part of `code-generator` since April 2019. Or do we rely on versions older than that?